pkg/kapp/config/default.go

```diff
@@ -577,6 +577,7 @@ changeRuleBindings:
 # Insert namespaces before all namespaced resources
 - rules:
   - "upsert after upserting change-groups.kapp.k14s.io/namespaces-{namespace}"
+  - "delete before deleting change-groups.kapp.k14s.io/namespaces-{namespace}"
```

Comment:
should we be only blocking namespace on service account deletion? (since that the sticking point)

Reply:
what is the problem in keeping it general?
I mean adding rule to delete all namaspaced resources before namespace itself

Now in current use case we have only SA is blocking but if in future something else will come then again we have to add rule for that. Keeping general will work for everything. Also logically all namespaced resources should get deleted before namespace.